lib/gitsh/tab_completion/dsl/option_transition_factory.rb

```diff
@@ -0,0 +1,60 @@
+require 'gitsh/tab_completion/automaton'
+require 'gitsh/tab_completion/matchers/unknown_option_matcher'
+
+module Gitsh
+  module TabCompletion
+    module DSL
+      class OptionTransitionFactory
+        def build(start_state, options = {})
+          Invocation.new(start_state, options).call
```

Comment:
I phrase it as "don't set ivars outside of the constructor", and `#build` is the constructor.